File: frontend/src/microservices/operations/airline/objects/CreateManagerFlightPlannerRequest.ts
// 本文件定义 `CreateManagerFlightPlanner` 的前端请求对象，用于航司管理员创建航班。

import type { ManagerCabinPricingInput } from './ManagerCabinPricingInput'

export type CreateManagerFlightPlannerRequest = {
  managerId: string
  flightNumber: string
  departureAirport: string
  arrivalAirport: string
  departureTime: string
  arrivalTime: string
  economyCabin: ManagerCabinPricingInput
  premiumEconomyCabin: ManagerCabinPricingInput
  businessCabin: ManagerCabinPricingInput
  firstCabin: ManagerCabinPricingInput
  currency: string
}

export const createManagerFlightPlannerRequestFromJson = (json: string): CreateManagerFlightPlannerRequest =>
  JSON.parse(json) as CreateManagerFlightPlannerRequest

export const createManagerFlightPlannerRequestToJson = (value: CreateManagerFlightPlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/operations/airline/objects/CreateManagerFlightPlannerRequest.scala
// 这个文件定义 airline 航班创建请求。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class CreateManagerFlightPlannerRequest(
    managerId: String,
    flightNumber: String,
    departureAirport: String,
    arrivalAirport: String,
    departureTime: String,
    arrivalTime: String,
    economyCabin: ManagerCabinPricingInput,
    premiumEconomyCabin: ManagerCabinPricingInput,
    businessCabin: ManagerCabinPricingInput,
    firstCabin: ManagerCabinPricingInput,
    currency: String
)
object CreateManagerFlightPlannerRequest:
  given sourceEncoder: Encoder[CreateManagerFlightPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[CreateManagerFlightPlannerRequest] = deriveDecoder
